console/v1/types_console_link.go

```diff
@@ -0,0 +1,41 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Extension for customizing OpenShift web console links
+type ConsoleLink struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ConsoleLinkSpec `json:"spec"`
+}
+
+type ConsoleLinkSpec struct {
+	Link `json:",inline"`
+	// menu determines which location in the console the link will be appended to
+	Location ConsoleLocationSelector `json:"location"`
+}
+
+// ConsoleMenuSelector is a set of possible menu targets to which a
+// link may be appended.  Links will be ordered alphabetically.
+type ConsoleLocationSelector string
```

Comment:
Should this be `ConsoleLinkLocationSelector`?

Reply:
Yeah that probably makes sense.